src/yadmark/benchmark/external/apfelpy_utils.py

```diff
@@ -17,6 +17,69 @@
 }
 
 
+def map_apfelpy_sf(init, observables, fns):
+    """Get required structure function.
+
+    Parameters
+    ----------
+    init: ap.initializers
+        Apfel++ initializers
+    observables: dict
+        observables runcard
+    fns: str
+        flavor number scheme
+
+    Returns
+    -------
+        Apfel++ structure function initalizer
+    """
+    MAP_ZM_CC = {
+        "F2m": init.InitializeF2CCMinusObjectsZM,
+        "FLm": init.InitializeFLCCMinusObjectsZM,
+        "F3m": init.InitializeF3CCMinusObjectsZM,
+        "F2p": init.InitializeF2CCPlusObjectsZM,
+        "FLp": init.InitializeFLCCPlusObjectsZM,
+        "F3p": init.InitializeF3CCPlusObjectsZM,
+    }
+
+    MAP_ZM_NC = {
+        "F2": init.InitializeF2NCObjectsZM,
+        "FL": init.InitializeFLNCObjectsZM,
+        "F3": init.InitializeF3NCObjectsZM,
+        "g1": init.Initializeg1NCObjectsZM,
+        "gL": init.InitializegLNCObjectsZM,
+        "g4": init.Initializeg4NCObjectsZM,
+    }
+
+    MAP_FFNS_NC = {
+        "F2": init.InitializeF2NCObjectsMassive,
+        "FL": init.InitializeFLNCObjectsMassive,
+    }
+    MAP_FFNS0_NC = {
+        "F2": init.InitializeF2NCObjectsMassiveZero,
+        "FL": init.InitializeFLNCObjectsMassiveZero,
+    }
+
+    if observables["prDIS"] == "CC":
+        projectile_pids = {
+            "electron": 11,
+            "positron": -11,
+            "neutrino": 12,
+            "antineutrino": -12,
+        }
+        if projectile_pids[observables["ProjectileDIS"]] > 0:
+            apfelpy_structure_functions = MAP_ZM_CC
+    else:
+        if fns == "ZM-VFNS":
+            apfelpy_structure_functions = MAP_ZM_NC
+        elif fns == "FFNS":
+            apfelpy_structure_functions = MAP_FFNS_NC
+        elif fns == "FFNS0":
+            apfelpy_structure_functions = MAP_FFNS0_NC
+
+    return apfelpy_structure_functions
```

Comment:
Small improvement to the code: early returns

```suggestion
        if projectile_pids[observables["ProjectileDIS"]] > 0:
            return MAP_ZM_CC
    else:
        if fns == "ZM-VFNS":
            return MAP_ZM_NC
        if fns == "FFNS":
            return MAP_FFNS_NC
        if fns == "FFNS0":
            return MAP_FFNS0_NC
```

Reply:
In principle, you could define the various maps in place (they are used only once), and they even have the wrong case, since they are not constants (they depend on `init`).

The other option would be to define them as top-level functions, that take `init` as a parameter and return the mapping.

Reply:
this actually reveals a hole in the definition: what happens if `projectile_pids[observables["ProjectileDIS"]] > 0` yields False?